src/TimingData.cpp

```diff
@@ -963,6 +997,9 @@ void TimingData::AddSegment( const TimingSegment *seg )
 	LOG->Trace( "AddSegment( %s )", TimingSegmentTypeToString(seg->GetType()).c_str() );
 	seg->DebugPrint();
 #endif
+	// NEET trick: class that will trigger the lookup table rebuild when it is
+	// destroyed.
+	lookup_updater_for_segment_add lookup_me_senpai(this);
```

Comment:
I may be mixing up programming languages here, but wouldn't this work better as a sort of static method instead of an instance trick?